### PR TITLE
Ian - Driver shift page

### PR DESCRIPTION
**Pull request by ru1nw**
In this PR, I added a page, [storybook](https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-4/prs/50/storybook/?path=/story/pages-shiftpage-shiftpage--default), and tests for driver shift page. ~~I also added the route to this page in App.js~~, and updated currentUserFixture.js from [Sophia's work](https://github.com/ucsb-cs156-s23/proj-gauchoride-s23-5pm-4/commit/059ec8fd830d952875ad93a5f06688c8d9d793ca) for testing purposes.
This PR is part of issue #8 of implementing the driver shift table spreadsheet. Closed PR #26 without merging due to the unforeseen scale of adding an entire page; will create more PRs to address the rest of the added files in #26.

Edit Jun 6 4:15PM: I removed the route that points to this page in App.js (revert all changes made by me) until the backend API works. The [storybook](https://ucsb-cs156-s23.github.io/proj-gauchoride-s23-5pm-4/prs/50/storybook/?path=/story/pages-shiftpage-shiftpage--default) still works as intended.
image

### CODE REVIEW

File: frontend/src/main/components/Shift/ShiftTable.js
Comment (SophiaWeiler) [inline frontend/src/main/components/Shift/ShiftTable.js:1]: I don't think you need this line

**Review by SophiaWeiler — CHANGES_REQUESTED** [review]
There is another line that disables the stryker that is not needed that I think you should remove

**Review by SophiaWeiler — DISMISSED** [review]
lgtm!

**Comment by ru1nw** [comment]
Planning to just merge this, skipping PR #42 since this PR builds on top of that. Will close #42 when this is merged.

**Review by zachmiller-ucsb — COMMENTED** [review]
We should wait to merge this until the endpoints for ShiftRepository are merged, so that ShiftPage does not give errors when accessing endpoints that do not exist yet. Similar issue as RideRequestEditPage.

**Review by SophiaWeiler — APPROVED** [review]
lgtm

**Review by pconrad — APPROVED** [review]
LGTM

**Pull request merged**